src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const setOfStates = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        setOfStates.push(currentState);
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        setOfStates.push(currentState);
+        break;
+      case 'removeProperties':
+        currentState = { ...currentState };
```

Comment:
Creating a copy of `currentState` is redundant here since you're already making a copy when you assign it with the spread operator in the previous cases. You can remove this line.

Reply:
I'm making a copy in the other cases, but if I don't make a copy in this one, it messes up one the object that resulted from the previous action, so if I remove it from here, 2 of the tests fail: Should handle multiple types and Should handle a long list of types